### PR TITLE
planner: add more test cases about dynamic-mode with new-collation

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: related to #24150 

Problem Summary: planner: add more test cases about dynamic-mode with new-collation

### What is changed and how it works?

planner: add more test cases about dynamic-mode with new-collation

### Check List 

Tests 

- Unit test

### Release note 

- planner: add more test cases about dynamic-mode with new-collation

### CODE REVIEW

File: executor/partition_table_test.go [hunk 1]
```diff
@@ -222,6 +223,46 @@ func (s *partitionTableSuite) TestPartitionInfoDisable(c *C) {
 	tk.MustQuery("select * from t_info_null where (date = '2020-10-02' or date = '2020-10-06') and app = 'xxx' and media = '19003006'").Check(testkit.Rows())
 }
 
+func (s *partitionTableSuite) TestNewCollation(c *C) {
```
Comment (wjhuang2016) [inline executor/partition_table_test.go:226]: It's not serial.

Comment (qw4990) [reply executor/partition_table_test.go]: You are right. Now I attach it to `testIntegrationSerialSuite` like other cases about collation.

**Review by rebelice — APPROVED** [review]
/lgtm

**Pull request merged**